pkg/hubble/observer/local_observer.go

```diff
@@ -95,9 +95,17 @@ func NewLocalServer(
 		"eventQueueSize": opts.MonitorBuffer,
 	}).Info("Configuring Hubble server")
 
+	maxFlows := opts.MaxFlows
+	if maxFlows > 1 && (maxFlows&(maxFlows-1) == 0) {
+		// The ring buffer has an actual capacity of 2^n-1. Thus, to avoid
+		// power of 2 specified capacity to be rounded up (eg 1024 -> 2047),
+		// subtract one to the provided capacity as this probably best
+		// reflects the actual caller's intent.
+		maxFlows -= 1
+	}
```

Comment:
nit: I think we should move this into `NewRing`

Reply:
I did that as I also think it's the right thing to do as also hinted by @glibsm in a comment above. However, the required changes are then more involved. As I would like to backport this change for v1.9, I propose to do this as part of another PR.